### ISSUE TITLE
About the system crash (data loss) while waking from hibernation/sleep

### ISSUE THREAD

**Issue by arshcaria** [darkhandz/XPS15-9550-Sierra#73]
I am looking into this problem right now. I've been using hackintosh for a week now but cannot reproduce this issue.

Is this issue already resolved? Or may I ask when this issue most often happens? Waking from hibernation or from sleep?

b.r.

**Comment by addisn** [comment]
You can try to set the hibernate mode to 0, which will prevent the machine to deep sleep(hibernate).
In my case, i haven't been corrupted after this procedure and so far so good.

**Comment by arshcaria** [comment]
Yes, I know that disabling hibernation is a workaround. I just would like to know if there is any certain routine to reproduce the issue.

**Comment by darkhandz** [comment]
@arshcaria I've no idea about this, but maybe you can take a look at[ this issue](https://github.com/syscl/XPS9350-macOS/issues/26)

**Comment by krim404** [comment]
this seems to fix the issue completly, but does not work on every machine:
https://github.com/wmchris/DellXPS15-9550-OSX/blob/master/4k_sector.md

**Comment by arshcaria** [comment]
@wmchris Thanks for the info. I noticed you mentioned "Samsung drives are incompatible". So drives like pm951 or sm951 cannot be reformatted to 4k sector?

**Comment by krim404** [comment]
you can try, but as far as i know the firmware of samsung is not offering this standard method.

**Comment by arshcaria** [comment]
Just tried that. `smartctl` does not recognize samsung nvme drives.

**Comment by krim404** [comment]
then you've missused it. they're recognizeable, they just dont offer the 4k option.

**Comment by arshcaria** [comment]
I ran `# smartctl -a /dev/nvme0` and got an error msg saying it could not identify the type of that device. I checked the `/dev` folder,  `nvme0` and its partitions devs are there. Any idea what is wrong? Thanks.

**Comment by solvar0719** [comment]
Hm, I wonder what the data loss looks like as it might be happening to me right now (on High Sierra). I am using xg3 which does support 4K sector (haven't converted it to 4K though) and after I got Clover to recognize my sleepimage on restoration, it simply goes straight into a reboot. Canceling hibernation doesn't make a difference either.

**Comment by krim404** [comment]
this is not the same issue. hibernation is saved in nvram. clear it.

**Comment by solvar0719** [comment]
@wmchris After a few hours of trial and error (and a painful restoration from Time Machine)  I found out that the culprit is the `NVRAM.plist` under the root directory. I deleted it and it booted, but I still haven't been able to get hibernation to work correctly. `HibernationFixup.kext` does make clover recognize hibernation but the system can't restore from it. To be honest I don't know how that `nvram.plist` got into the root directory in the first place.

**Comment by krim404** [comment]
nvram.plist is from emuvariable clover extension to emulate a physical nvram. hibernation is not a good idea on this machine at all. possibility for data corruption etc.

normally hibernation is not needed, because sleep power consumption was less than 2% per day...
